OpenRA.Mods.Common/SpriteLoaders/ShpTSLoader.cs

```diff
@@ -118,7 +118,9 @@ bool IsShpTS(Stream s)
 			while (w == 0 && h == 0 && f++ < imageCount);
 
 			s.Position = start;
-			return type < 4;
+
+			// Type 18 is a special case for RA2 low bridges.
```

Comment:
https://github.com/OpenRA/OpenRA/blob/287290bbdf08ac1ec884213ef1612a185242c2a1/OpenRA.Mods.Common/SpriteLoaders/ShpTSLoader.cs#L108-L118

A quick check against ra2ff.txt confirms that this is completely bogus, and tries to parse unrelated data as frame width/height/compression types.  This loop needs to be rewritten to account for the actual layout of the image header.